### PR TITLE
OJ-2609: Audit alarms review

### PR DESCRIPTION
**Pull request by mikebeeby**
### What changed

- Removed dimension from PutEventAlarm as it did not exist on the metric, now PutEvents Failures to any EventBus will alarm, the only way I could get this to work.
- Refactored conditions on all audit alarms to actually adhere to the alarm description.
- Set Audit alarms TreatMissingData to notBreaching, otherwise it would be strange for the alarm to transition from InsufficientData -> Ok upon 1-3 errors within an hour.

### Why did it change

I could not see the EventBusName dimension on the putEventsApproximateFailedCount metric, so I have removed to keep the alarm for all PutEvents to any EventBus.
The `TreatMissingData: missing` behaved differently then it looks like was thought with how the alarms are configured. There is a good description on this [page](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html), the 2 tables about half way down the page. I believe these alarms will now meet the intended behaviour from the description. 

### Issue tracking

- [OJ-2609](https://govukverify.atlassian.net/browse/OJ-2609)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2609]: https://govukverify.atlassian.net/browse/OJ-2609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by liamStockwellGDS — DISMISSED** [review]

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1853,19 +1853,16 @@ Resources:
...
+      AlarmDescription: !Sub "${AWS::StackName}-${Environment} PutEvents (PutEvents Failed) 4 or more requests in the last hour. ${SupportManualURL}"
       AlarmName: !Sub "${AWS::StackName}-${Environment}-CheckHmrcEventBus-PutEventsApproximateFailedCount-alarm"
       MetricName: PutEventsApproximateFailedCount
       ComparisonOperator: GreaterThanThreshold
       Namespace: "AWS/Events"
       Statistic: Sum
-      DatapointsToAlarm: 4
-      EvaluationPeriods: 12
-      Dimensions:
-        - Name: EventBusName
```
Comment (surajkumar) [inline infrastructure/template.yaml:1865]: Let's discuss the removal of this, I'm not convinced we should remove it as I can see the filter in the console

Comment (mikebeeby) [reply infrastructure/template.yaml:1865]: Hmm I'll have another look but the actual metrics do not seem to have the EventBusName set as a dimension.

image


You can see the PutEventsApproximateCallCount metric also does not.

image

Comment (jkunle) [reply infrastructure/template.yaml:1865]: @mikebeeby have a lool at https://aws.amazon.com/blogs/compute/enhanced-amazon-cloudwatch-metrics-for-amazon-eventbridge/

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -1853,19 +1853,16 @@ Resources:
       AlarmActions:
         - !ImportValue platform-alarm-warning-alert-topic
       InsufficientDataActions: []
-      AlarmDescription: !Sub "${CheckHmrcEventBus} PutEvents (PutEvents Failed) 4 or more requests in the last hour. ${SupportManualURL}"
+      AlarmDescription: !Sub "${AWS::StackName}-${Environment} PutEvents (PutEvents Failed) 4 or more requests in the last hour. ${SupportManualURL}"
```
Comment (jkunle) [inline infrastructure/template.yaml:1856]: If we are making changes to Alarms and descriptions we should remember to make changes here as https://team-manual.account.gov.uk/teams/CRI-Orange-team/supporting-cri-orange/hmrc-nino-runbook/#backend-api-alarms

Comment (jkunle) [reply infrastructure/template.yaml:1856]: All the descriptions no longer tally with `4 or more requests in the last hour`

File: infrastructure/template.yaml (hunk 1 above)
Comment (mikebeeby) [reply infrastructure/template.yaml:1865]: Ahh nice link, thanks better than anything I have found googling this morning! 
So yeah no dimensions on the PutEventsApproximateFailedCount metric.

File: infrastructure/template.yaml (hunk 2 above)
Comment (mikebeeby) [reply infrastructure/template.yaml:1856]: Yep I'll raise a PR once/if these changes are approved.

Unsure what you mean with the last comment?

**Pull request merged**